ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh

```diff
@@ -10,16 +10,38 @@ az --version
 
 # set the parameters we'll need as env vars
 AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
-AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
-AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
-AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
 
 # log in with az
 if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
     az cloud set --name AzureUSGovernment
+elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+    if [ ! -f "${CLUSTER_PROFILE_DIR}/cloud_name" ]; then
+        echo "Unable to get specific ASH cloud name!"
+        exit 1
+    fi
+    cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+
+    AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
```

Comment:
The line is the same as the line 12, right?

Reply:
sorry, should be `AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"`.
On ASH, as service principal file for dev ASH PPE is not saved into `CLUSTER_PROFILE_DIR`, but mounted as secrets under namespace test-credentials([here](https://github.com/openshift/release/blob/master/ci-operator/step -registry/ipi/conf/azurestack/ipi-conf-azurestack-commands.sh#L28)), then move it into `${SHARED_DIR}`.

Do we need to consider PPE env when updating some azure provision step to support ASH? If not,line42~line44 can be moved back.

Reply:
Ah, got it. I think we can only take wwt ash into consideration for the steps only used in QE's wwt ash environment.

Reply:
okay, will move line42~line44 back to original places

Reply:
updated